src/javascripts/components/cookies-page.js

```diff
@@ -0,0 +1,167 @@
+  if ($errors.length !== 0) {
+    this.showErrorSummary($errors)
+    return
+  }
+
+  // otherwise save preferences to cookie and show success notification
+  setConsentCookie(preferences)
+  this.hideErrorSummary()
+  this.showSuccessNotification()
+}
+
+CookiesPage.prototype.showUserPreference = function ($cookieFormGroup, preferences) {
+  if (!preferences) {
+    return
+  }
+
+  var cookieType = this.getCookieType($cookieFormGroup)
+
+  if (!cookieType || preferences[cookieType] === undefined) {
+    return
+  }
+
+  var preference = preferences[cookieType]
+
+  if (preference === true) {
+    $cookieFormGroup.querySelector('input[value="yes"]').checked = true
+  } else if (preference === false) {
+    $cookieFormGroup.querySelector('input[value="no"]').checked = true
+  }
+}
+
+/* Hide an error message for a cookie option. */
+CookiesPage.prototype.hideErrorMessage = function ($cookieFormGroup) {
+  var errorMessage = $cookieFormGroup.querySelector('.govuk-error-message')
+  if (errorMessage !== null) {
+    errorMessage.parentNode.removeChild(errorMessage)
+  }
+
+  var errorMessageHighlight = $cookieFormGroup.querySelector('.govuk-form-group--error')
+  errorMessageHighlight && errorMessageHighlight.classList.remove('govuk-form-group--error')
+}
+
+/* Show an error message for a cookie option.
+ *
+ * If the user has not selected yes or no to accept a cookie type, we want to
+ * show an error message.
+ */
+CookiesPage.prototype.showErrorMessage = function ($cookieFormGroup) {
+  // Do not show error message more than once
+  if ($cookieFormGroup.querySelector('.govuk-error-message')) {
+    return
+  }
+
+  var cookieType = this.getCookieType($cookieFormGroup)
+
+  // error message is defined in error summary macro call in template
+  var errorMessageText = this.$errorSummary.querySelector('a[href="#' + cookieType + '"]').innerText
+
+  var formGroup = $cookieFormGroup.querySelector('.govuk-form-group')
+  formGroup && formGroup.classList.add('govuk-form-group--error')
+
+  var errorMessageId = cookieType + '-error'
+  var errorMessageSpan = document.createElement('span')
+  errorMessageSpan.id = errorMessageId
+  errorMessageSpan.className = 'govuk-error-message'
+  errorMessageSpan.innerHTML = '<span class="govuk-visually-hidden">Error:</span> ' + errorMessageText
+
+  var siblingElement = $cookieFormGroup.querySelector('.govuk-radios')
+  var parentElement = siblingElement.parentElement
+  parentElement.insertBefore(errorMessageSpan, siblingElement)
+
+  var fieldset = $cookieFormGroup.querySelector('.govuk-fieldset')
+  fieldset && fieldset.setAttribute('aria-describedby', errorMessageId)
+}
+
+CookiesPage.prototype.hideErrorSummary = function () {
+  this.$errorSummary.setAttribute('hidden', '')
+}
+
+CookiesPage.prototype.showErrorSummary = function ($errors) {
+  this.$errorSummary.removeAttribute('hidden')
+
+  this.$errorSummary.firstElementChild.focus()
```

Comment:
Again this all feels a little strange because we're calling something the error summary, but it's actually a wrapper around the error summary.

Unlike the form group, it's possible to set attributes on the error summary and notification banner, so I'm wondering if we could simplify things by adding the `js-` class and the hidden attribute to the component itself, and lose the wrapper?